### PR TITLE
operator: add ownerReferences to generated resources

### PR DESCRIPTION
**Pull request by rokrokss**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

Some opensource tools like ArgoCD has dependencies to the `ownerReferences` metadata for resources created in operator-pattern.

### CODE REVIEW

**Review by morvencao — CHANGES_REQUESTED** [review]
If you check the kubernetes GC doc, you will find that there are some rules to set the owner references:

> Cross-namespace owner references are disallowed by design. This means:
>   Namespace-scoped dependents can only specify owners in the same namespace, and owners that are cluster-scoped.
     Cluster-scoped dependents can only specify cluster-scoped owners, but not namespace-scoped owners.

For istio operator, the iop instance can be deployed to any namespace, which may be different from  the namespace that istio control plane is deployed, so we can't set owner reference to the IOP instance.

Besides, for cluster-scoped resources, eg. clusterrole, clusterrolebing, we can't set owner reference to the IOP instance, because IOP instance is namespace-scoped resource.

**Comment by rokrokss** [comment]
@morvencao by this PR, generated istio resources' ownerRef is tagged with IstioOperator CRD, not the iop instance.

File: operator/pkg/helmreconciler/reconciler.go [hunk 1]
```diff
@@ -377,8 +379,29 @@ func (h *HelmReconciler) getOwnerLabels(componentName string) (map[string]string
...
+		return nil, err
+	}
+	ownerRef := &metav1.OwnerReference{
+		APIVersion:         h.iop.ApiVersion,
+		Kind:               h.iop.Kind,
+		Name:               crName,
+		UID:                crUID,
+		BlockOwnerDeletion: pointer.BoolPtr(true),
+		Controller:         pointer.BoolPtr(true),
+	}
```
Comment (morvencao) [inline operator/pkg/helmreconciler/reconciler.go:399]: Owner ref is IOP instance, right?

Comment (rokrokss) [reply operator/pkg/helmreconciler/reconciler.go:399]: I understood by iop instance you meant operator deployment, but I get it now.

**Comment by rokrokss** [comment]
@morvencao Thanks

> Namespace to install control plane resources into. If unset, Istio will be installed into the same namespace as the IstioOperator CR.

Yes, you are right. the resources can be deployed to other namespaces than the CR

**Comment by rokrokss** [comment]
I guess I need to implement feature for handling istio resources in ArgoCD code.

**Pull request closed without merging**